src/pal/src/configure.cmake

```diff
@@ -43,11 +43,17 @@ if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
 endif()
 
 if(NOT CMAKE_SYSTEM_NAME STREQUAL FreeBSD AND NOT CMAKE_SYSTEM_NAME STREQUAL NetBSD)
```

Comment:
Setting the dl libs is not needed for the check_cxx_source_compiles - it just compiles the source and doesn't try to link it.

Reply:
Other than this, it LGTM.

Reply:
Clang failed during the configuration and when I checked CMakeError.log, I found:

```cmake
int main(void){return 0;}

Performing C++ SOURCE FILE Test HAVE_LTTNG_TRACEPOINT_H failed with the following output:
Change Dir: /coreclr/bin/obj/Linux.x64.Checked/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_b8a86/fast"
/usr/bin/make -f CMakeFiles/cmTC_b8a86.dir/build.make CMakeFiles/cmTC_b8a86.dir/build
make[1]: Entering directory '/coreclr/bin/obj/Linux.x64.Checked/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_b8a86.dir/src.cxx.o
/usr/bin/clang++  -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_FILE_OFFSET_BITS=64 -D_POSIX_C_SOURCE=200809L -D_SVID_SOURCE  -Wall -Wno-null-conversion -std=c++11 -Wno-invalid-offsetof -DHAVE_LTTNG_TRACEPOINT_H   -o CMakeFiles/cmTC_b8a86.dir/src.cxx.o -c /coreclr/bin$
Linking CXX executable cmTC_b8a86
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_b8a86.dir/link.txt --verbose=1
/usr/bin/clang++  -Wall -Wno-null-conversion -std=c++11 -Wno-invalid-offsetof -DHAVE_LTTNG_TRACEPOINT_H    -rdynamic CMakeFiles/cmTC_b8a86.dir/src.cxx.o  -o cmTC_b8a86
CMakeFiles/cmTC_b8a86.dir/src.cxx.o: In function `__tracepoints__init()':
src.cxx:(.text+0x84): undefined reference to `dlopen'
CMakeFiles/cmTC_b8a86.dir/src.cxx.o: In function `__tracepoint__init_urcu_sym()':
src.cxx:(.text+0x10e): undefined reference to `dlsym'
src.cxx:(.text+0x147): undefined reference to `dlsym'
src.cxx:(.text+0x180): undefined reference to `dlsym'
CMakeFiles/cmTC_b8a86.dir/src.cxx.o: In function `__tracepoints__destroy()':
src.cxx:(.text+0x222): undefined reference to `dlclose'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
CMakeFiles/cmTC_b8a86.dir/build.make:97: recipe for target 'cmTC_b8a86' failed
make[1]: *** [cmTC_b8a86] Error 1
make[1]: Leaving directory '/coreclr/bin/obj/Linux.x64.Checked/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_b8a86/fast' failed
make: *** [cmTC_b8a86/fast] Error 2
```

Reply:
Hmm, you are right. I am sorry, I was quite sure I've tested it locally.

Reply:
Actually, one last nit - now that you use the CMAKE_DL_LIBS, we don't need to have the `if`s around setting and clearing the CMAKE_REQUIRED_LIBRARIES. The variable is set to empty on systems like FreeBSD / NetBSD that don't have the `dl` libs.